### ISSUE TITLE
Making non-ambient typings for default function commonjs package

### ISSUE THREAD

**Issue by jonaskello** [Microsoft/TypeScript#11732]
Now sure if this is a bug or by-design, I tried [stackoverflow](http://stackoverflow.com/questions/39873009/making-non-ambient-typings-for-default-function-commonjs-package) but no answers so here it goes:

I am trying to make non-ambient typings for [chokidar-socket-emitter](https://www.npmjs.com/package/chokidar-socket-emitter) using tsc 2.0.3.

Here is my attempt:

**test.ts**:

```
    import chokidar from 'chokidar-socket-emitter';

    chokidar({
        app: null
    });
```

**node_modules/@types/chokidar-socket-emitter/index.d.ts**:

```
export default function chokidarEvEmitter(options: any);
```

**tsconfig.json**:

```
    {
        "compilerOptions": {
            "module": "commonjs",
            "target": "es5",
            "noImplicitAny": false,
            "sourceMap": false
        }
    }
```

**package.json**:

```
    {
      "name": "typing_default",
      "version": "1.0.0",
      "dependencies": {
        "chokidar-socket-emitter": "^0.5.4"
      }
    }
```

Running `tsc` whill produce this output:

**test.js**:

```
    "use strict";
    var chokidar_socket_emitter_1 = require("chokidar-socket-emitter");
    chokidar_socket_emitter_1.default({
        app: null
    });
```

When I then run `node test.js` I get:

```
TypeError: chokidar_socket_emitter_1.default is not a function
```

The problem is that an extra `.default` is emitted at `chokidar_socket_emitter_1.default(...)`. If I just remove `.default` it works OK.

My question is:

Is there some way I can make tsc not emit this extra .default and still use non-ambient typings and regular ES6 imports?

**Comment by mhegazy** [comment]
`chokidar-socket-emitter`, and any other commonjs module, would not have a `default` export. [`default` exports](http://www.typescriptlang.org/docs/handbook/modules.html#default-exports) are ES6 concept.

you want to define the module using `export =`. i.e.:

```ts
declare function chokidarEvEmitter(options: any);
export = chokidarEvEmitter;
```

then use it as:

```ts
import chokidar  = require('chokidar-socket-emitter');
```

**Comment by jonaskello** [comment]
Thanks for info! I tried that syntax too, but then I had trouble with the tsc option `"target": "es6"` which seems to be incompatible with the `import = require` syntax. Is there some way to provide typings for commonjs modules while still using es6 as target?

**Comment by mhegazy** [comment]
for this specific case. i do not know... there is no ES6 module implementation. so you are using babel to transpile it. if this is the case, then use `--allowSyntheticDefaultImports`. this will allow you to import the module as a default, since babel will rewrite the default import for you.

so the .d.ts file should still look like:

```ts
declare function chokidarEvEmitter(options: any);
export = chokidarEvEmitter;
```

the import would be:

```ts
  import chokidar from 'chokidar-socket-emitter';
```

compile it with:
```cmd
tsc --target ES2015 --allowSyntheticDefaultImports
```

**Comment by blakeembrey** [comment]
@jonaskello You can also do `target: es6` and `module: commonjs`. That's what I use, assuming you aren't relying on actual ES6 modules being emitted.

**Comment by jonaskello** [comment]
I was actually using `target: es6` with `module: commonjs`. I'm trying to use a common `tsconfig.json` for both client and server part in my app. In development I load the *.ts files directly into Chrome with SystemJS and on the server-side I run *.ts files directly with ts-node. Both Chrome and node supports es6 syntax so I don't think`target: es6` should not be a problem in itself. I will try with the --allowSyntheticDefaultImports and see if that helps. If not I'll try to make a small repro to show what I'm encountering.

**Comment by aluanhaddad** [comment]
@jonaskello It is worth noting that the behavior of `allowSyntheticDefaultImports` is to inform the compiler that it should perceive a module declared as
```ts
export = class {};
```
as having the alternate shape
```ts
export default class {};
```
That is it _views_ the module in both ways from a type checking point of view but does not in any way alter its output. This is specifically to allow code that relies on an ES6 <- CJSinterop aware environment, such as Babel or SystemJS by default, to typecheck correctly.
Basically, if
```ts
import chokidar from 'chokidar-socket-emitter';
```
binds `chokidar` to the value of the target `module.exports` then your runtime is "lifting" it into a binding named default on the module record, if you are comfortable with that behavior, you should specify `allowSyntheticDefaultImports` but if you are not comfortable with it, you should disable this behavior in your runtime.
